Assets/Python/Stability.py

```diff
@@ -760,23 +758,23 @@ def calculateStability(iPlayer):
 				iRecentlyConquered += 1
 			
 		# Religions
-		bNonStateReligion = False
-		for iReligion in range(iNumReligions):
-			if iReligion != iStateReligion and city.isHasReligion(iReligion):
-				if not isTolerated(iPlayer, iReligion) and not gc.getReligionInfo(iReligion).isLocal():
-					bNonStateReligion = True
-					break
-				
-		if city.isHasReligion(iStateReligion):
-			iStateReligionPopulation += city.getPopulation()
-			if not bNonStateReligion: iOnlyStateReligionPopulation += city.getPopulation()
-				
-		if bNonStateReligion: 
-			if iStateReligion >= 0 and city.isHasReligion(iStateReligion): iDifferentReligionPopulation += city.getPopulation() / 2
-			else: iDifferentReligionPopulation += city.getPopulation()
-			
 		if city.getReligionCount() == 0:
-			iNoReligionPopulation += city.getPopulation()
+			iNoReligionPopulation += iPopulation
+		else:
+			bNonStateReligion = False
+			for iReligion in range(iNumReligions):
+				if iReligion != iStateReligion and city.isHasReligion(iReligion):
+					if not isTolerated(iPlayer, iReligion) and not gc.getReligionInfo(iReligion).isLocal():
+						bNonStateReligion = True
+						break
+					
+			if city.isHasReligion(iStateReligion):
+				iStateReligionPopulation += iPopulation
+				if not bNonStateReligion: iOnlyStateReligionPopulation += iPopulation
+					
+			if bNonStateReligion: 
+				if iStateReligion >= 0 and city.isHasReligion(iStateReligion): iDifferentReligionPopulation += iPopulation / 2
+				else: iDifferentReligionPopulation += iPopulation
 		

```

Comment:
It now skippes code if the city does not have a religion.

Minor note: I a civ has a pagan state religion, the variable iStateReligionPopulation now doesn't increase anymore. This variable is only relevant if the civ does not have a pagan state religion, so this does minor change does not matter.

Reply:
I agree, iStateReligionPopulation does not matter without a real state religion.